In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as st

In [ ]:
n=80 #sample size

In [ ]:
df=pd.read_csv('Ajio.csv')

In [ ]:
df

Product_URL                 Brand  \
0     https://www.ajio.com/netplay-checked-polo-t-sh...               netplay   
1     https://www.ajio.com/netplay-tapered-fit-flat-...               netplay   
2     https://www.ajio.com/the-indian-garage-co-stri...  the-indian-garage-co   
3     https://www.ajio.com/performax-heathered-crew-...             performax   
4     https://www.ajio.com/john-players-jeans-washed...    john-players-jeans   
...                                                 ...                   ...   
1004  https://www.ajio.com/wild-west-printed-polo-t-...             wild-west   
1005  https://www.ajio.com/dnmx-typographic-print-cr...                  dnmx   
1006  https://www.ajio.com/the-indian-garage-co-leaf...  the-indian-garage-co   
1007  https://www.ajio.com/u-s-polo-assn-graphic-pri...        u-s-polo-assn-   
1008  https://www.ajio.com/u-s-polo-assn-geometric-p...        u-s-polo-assn-   

                                   Description    Id_Product  \
0                         Checked Polo T-shirt  4.411370e+11   
1              Tapered Fit Flat-Front Trousers  4.411240e+11   
2     Striped Slim Fit Shirt with Patch Pocket  4.604540e+11   
3                  Heathered Crew-Neck T-shirt  4.410370e+11   
4        Washed Skinny Fit Jeans with Whiskers  4.411290e+11   
...                                        ...           ...   
1004                      Printed Polo T-shirt  4.640420e+11   
1005       Typographic Print Crew-Neck T-shirt  4.411190e+11   
1006                 Leaf Print Slim Fit Shirt  4.605530e+11   
1007           Graphic Print Crew-Neck T-shirt  4.690800e+11   
1008   Geometric Print Shirt with Patch Pocket  4.690800e+11   

                                              URL_image Category_by_gender  \
0     https://assets.ajio.com/medias/sys_master/root...                Men   
1     https://assets.ajio.com/medias/sys_master/root...                Men   
2     https://assets.ajio.com/medias/sys_master/root...                Men   
3     https://assets.ajio.com/medias/sys_master/root...                Men   
4     https://assets.ajio.com/medias/sys_master/root...                Men   
...                                                 ...                ...   
1004  https://assets.ajio.com/medias/sys_master/root...                Men   
1005  https://assets.ajio.com/medias/sys_master/root...                Men   
1006  https://assets.ajio.com/medias/sys_master/root...                Men   
1007  https://assets.ajio.com/medias/sys_master/root...                Men   
1008  https://assets.ajio.com/medias/sys_master/root...                Men   

     Discount Price (in Rs.) Original Price (in Rs.)     Color  
0                        559                     699     white  
1                        720                   1,499      navy  
2                        495                   1,649     white  
3                        329                     599  charcoal  
4                        899                     999  jetblack  
...                      ...                     ...       ...  
1004                     689                     999      navy  
1005                     319                     399      navy  
1006                     483                   2,099     white  
1007                     588                   1,199      teal  
1008                   1,039                   1,999       red  

[1009 rows x 9 columns]

In [ ]:
df.describe()


Id_Product
count  1.009000e+03
mean   4.485616e+11
std    1.232813e+10
min    4.102240e+11
25%    4.411200e+11
50%    4.411300e+11
75%    4.608860e+11
max    4.691950e+11

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 9 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Product_URL              1009 non-null   object 
 1   Brand                    1009 non-null   object 
 2   Description              1009 non-null   object 
 3   Id_Product               1009 non-null   float64
 4   URL_image                1009 non-null   object 
 5   Category_by_gender       1009 non-null   object 
 6   Discount Price (in Rs.)  1009 non-null   object 
 7   Original Price (in Rs.)  1009 non-null   object 
 8   Color                    1009 non-null   object 
dtypes: float64(1), object(8)
memory usage: 71.1+ KB


In [ ]:
df['Discount Price (in Rs.)'] = df['Discount Price (in Rs.)'].str.strip(',')
df['Discount Price (in Rs.)'] = pd.to_numeric(df['Discount Price (in Rs.)'],errors='coerce')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 9 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Product_URL              1009 non-null   object 
 1   Brand                    1009 non-null   object 
 2   Description              1009 non-null   object 
 3   Id_Product               1009 non-null   float64
 4   URL_image                1009 non-null   object 
 5   Category_by_gender       1009 non-null   object 
 6   Discount Price (in Rs.)  834 non-null    float64
 7   Original Price (in Rs.)  1009 non-null   object 
 8   Color                    1009 non-null   object 
dtypes: float64(2), object(7)
memory usage: 71.1+ KB


In [ ]:
m=df['Discount Price (in Rs.)'].mean()
m #population mean

574.6642685851319

Z - Test

Null Hypothesis      : H0 : Average discount price is 575 rupees (m==sm)
Alternate Hypothesis : H1 : Average discount price is not 575 rupees (m!=sm)

In [ ]:
sample = df.sample(n)
sm=sample['Discount Price (in Rs.)'].mean()
sm #sample mean

589.9852941176471

In [ ]:
sd=sample['Discount Price (in Rs.)'].std()
sd #standard deviation

187.37316049276248

In [ ]:
alpha=0.05 #test statistics
z= (sm-m)/(sd/np.sqrt(n))
z

0.7313503062150046

In [ ]:
z_critical = st.norm.ppf(alpha/2)
z_critical

-1.9599639845400545

In [ ]:
if (z>abs(z_critical)): # two tail test
  print('Rejecting Null Hypothesis')
else:
  print('Fail to Reject Null Hypothesis')

Fail to Reject Null Hypothesis
